integration/presto/src/main/prestodb/org/apache/carbondata/presto/readers/SliceStreamReader.java

```diff
@@ -124,6 +124,17 @@ public void putByteArray(int rowId, int count, byte[] value) {
     }
   }
 
+  @Override
+  public void putAllByteArray(byte[] data, int offset, int length) {
+    int[] lengths = getLengths();
+    int[] offsets = getOffsets();
+    for (int i = 0; i < lengths.length; i++) {
+      if (offsets[i] != 0) {
```

Comment:
Is this check required ? It is LV encoded byte array. Even offsets[0] wouldn't be 0 as it would point to offset of value field.

Reply:
offset values were 0 by default, due to which after select- empty rows were showed. Also select count(*) showed wrong results. Like you said offset can never be 0. If it is 0 then put should not be called. Hence this check.